app/views/shared/comments/_comments.html.erb

```diff
@@ -12,15 +12,16 @@
     <% end %>
     <div class="comments-list">
       <%= render partial: 'shared/comments/list.html.erb', locals: { comments: comments} %>
+      <% comments.mark_as_seen_by(current_user) %>
```

Comment:
Isn't this a bit too much logic for the view?

Reply:
It is directly connected to the view. Comments should be marked only when they are rendering and these changes affect only representation.